src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -1,348 +1,72 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Collections.Immutable;
-using System.Diagnostics;
-using System.Reflection;
-using System.Runtime.CompilerServices;
+#if !NET
+using System.Text.RegularExpressions;
+#endif
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.CSharp.Formatting;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.Editing;
-using Microsoft.CodeAnalysis.Formatting;
-using Microsoft.CodeAnalysis.Options;
-using Microsoft.CodeAnalysis.Simplification;
 using Microsoft.DotNet.ApiSymbolExtensions;
 using Microsoft.DotNet.ApiSymbolExtensions.Filtering;
 using Microsoft.DotNet.ApiSymbolExtensions.Logging;
-using Microsoft.DotNet.GenAPI.SyntaxRewriter;
 
 namespace Microsoft.DotNet.GenAPI
 {
     /// <summary>
     /// Processes assembly symbols to build corresponding structures in C# language.
     /// </summary>
-    public sealed class CSharpFileBuilder : IAssemblySymbolWriter, IDisposable
+    public sealed class CSharpFileBuilder : IAssemblySymbolWriter
```

Comment:
I extracted all the CSharpFileBuilder logic that visited nodes and put it in a separate class calls CSharpAssemblyVisitor (and its corresponding IAssemblyVisitor, in case we ever do the same for VB). I did this because that's the code I need to share with ApiDiff.

The only code left in CSharpFileBuilder is the code that writes the assembly to disk (after calling the necessary CSharpAssemblyVisitor methods).